FirebaseMessaging/Tests/UnitTests/FIRMessagingCheckinServiceTest.m

```diff
@@ -0,0 +1,158 @@
+                      completion:^(FIRMessagingCheckinPreferences *checkinPreferences,
+                                   NSError *error) {
+                        XCTAssertNil(error);
+                        XCTAssertEqualObjects(checkinPreferences.deviceID, kDeviceAuthId);
+                        XCTAssertEqualObjects(checkinPreferences.versionInfo, kVersionInfo);
+                        // For accuracy purposes it's better to compare seconds since the test
+                        // should never run for more than 1 second.
+                        NSInteger expectedTimestampInSeconds =
+                            (NSInteger)FIRMessagingCurrentTimestampInSeconds();
+                        NSInteger actualTimestampInSeconds =
+                            checkinPreferences.lastCheckinTimestampMillis / 1000.0;
+                        XCTAssertEqual(expectedTimestampInSeconds, actualTimestampInSeconds);
+                        XCTAssertTrue([checkinPreferences hasValidCheckinInfo]);
+                        [checkinCompletionExpectation fulfill];
+                      }];
+
+  [self waitForExpectationsWithTimeout:5
+                               handler:^(NSError *error) {
+                                 XCTAssertNil(error);
+                               }];
+}
+
+- (void)testFailedCheckinService {
+  [FIRMessagingCheckinService setGlobalTestBlock:[self failCheckinCompletionHandler]];
+
+  XCTestExpectation *checkinCompletionExpectation =
+      [self expectationWithDescription:@"Checkin Completion"];
+
+  [self.checkinService
+      checkinWithExistingCheckin:nil
+                      completion:^(FIRMessagingCheckinPreferences *preferences, NSError *error) {
+                        XCTAssertNotNil(error);
+                        XCTAssertNil(preferences.deviceID);
+                        XCTAssertNil(preferences.secretToken);
+                        XCTAssertFalse([preferences hasValidCheckinInfo]);
+                        [checkinCompletionExpectation fulfill];
+                      }];
+
+  [self waitForExpectationsWithTimeout:5
+                               handler:^(NSError *error) {
+                                 if (error) {
+                                   XCTFail(@"Checkin Timeout Error: %@", error);
+                                 }
+                               }];
+}
+
+#pragma mark - Stub
+
+- (FIRMessagingCheckinPreferences *)stubCheckinCacheWithValidData {
+  NSDictionary *gservicesData = @{
+    @"FIRMessagingVersionInfo" : kVersionInfo,
+    @"FIRMessagingLastCheckinTimestampKey" : @(FIRMessagingCurrentTimestampInMilliseconds())
+  };
+  FIRMessagingCheckinPreferences *checkinPreferences =
+      [[FIRMessagingCheckinPreferences alloc] initWithDeviceID:kDeviceAuthId
+                                                   secretToken:kSecretToken];
+  [checkinPreferences updateWithCheckinPlistContents:gservicesData];
+  return checkinPreferences;
+}
+
+#pragma mark - Swizzle
```

Comment:
Is it still relevant? I didn't find swizzling below.

Reply:
This should be resolved after the merging.